### PR TITLE
Extend authorization benchmark

### PR DESCRIPTION
**Pull request by wojtek-t**
```release-note
NONE
```

### CODE REVIEW

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go [hunk 1]
```diff
@@ -732,6 +735,23 @@ func BenchmarkAuthorization(b *testing.B) {
...
 	}
 
+	podToAdd := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: "testwrite", Namespace: "ns0"},
+		Spec: corev1.PodSpec{
+			NodeName:           "node0",
+			ServiceAccountName: "default",
+		},
+	}
+	for i := 0; i < opts.sharedSecretsPerPod; i++ {
```
Comment (liggitt) [inline plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: should we populate all the `*PerPod` dimensions here? could extract the pod-specific bits of `generate()` to `generatePod(node, namespace, pod string, opts sampleDataOpts) (*corev1.Pod, []*corev1.PersistentVolume, []*storagev1.VolumeAttachment)` and call that directly here to construct the pod

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go [hunk 2]
```diff
@@ -521,17 +522,17 @@ func TestAuthorizerSharedResources(t *testing.T) {
...
+	nodes       int
+	namespaces  int
 	podsPerNode int
 
 	attachmentsPerNode int
 
-	sharedConfigMapsPerPod int
-	sharedSecretsPerPod    int
-	sharedPVCsPerPod       int
+	sharedConfigMapsPerNamespace int
```
Comment (liggitt) [inline plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: it's not immediately obvious what this controls... godoc? same for sharedSecretsPerNamespace

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go [hunk 3]
```diff
@@ -840,6 +851,15 @@ func populate(graph *Graph, nodes []*corev1.Node, pods []*corev1.Pod, pvs []*cor
 	}
 }
 
+func randomSubset(a, b int) []int {
+	result := make([]int, b)
+	for i := 0; i < b; i++ {
+		result[i] = i
+	}
+	rand.Shuffle(b, func(i, j int) { result[i], result[j] = result[j], result[i] })
```
Comment (mikedanese) [inline plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: this is equivalent to `result := rand.Perm(b)`

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go [hunk 4]
```diff
@@ -840,6 +851,15 @@ func populate(graph *Graph, nodes []*corev1.Node, pods []*corev1.Pod, pvs []*cor
...
+	return result[:a]
+}
```
Comment (mikedanese) [inline plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: This is equivalent to:
```
func randomSubset(a, b int) []int {
	return rand.Perm(b)[:a]
}
```

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go [hunk 5]
```diff
@@ -644,16 +645,18 @@ func BenchmarkAuthorization(b *testing.B) {
...
+		podsPerNode:                  200,
+		attachmentsPerNode:           20,
+		sharedConfigMapsPerNamespace: 0,
+		sharedConfigMapsPerPod:       0,
+		uniqueConfigMapsPerPod:       1,
+		sharedSecretsPerNamespace:    1,
+		sharedSecretsPerPod:          1,
+		uniqueSecretsPerPod:          1,
+		sharedPVCsPerPod:             0,
+		uniquePVCsPerPod:             1,
```
Comment (mikedanese) [inline plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: We should also add service accounts to this at some point.

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go (hunk 4 above)
Comment (wojtek-t) [reply plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: done

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go (hunk 1 above)
Comment (wojtek-t) [reply plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: done

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go (hunk 2 above)
Comment (wojtek-t) [reply plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: done

**Review by wojtek-t — COMMENTED** [review]
Thanks for the review - PTAL

**Comment by wojtek-t** [comment]
@liggitt - friendly ping

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go [hunk 6]
```diff
@@ -639,21 +651,23 @@ func BenchmarkWriteIndexMaintenance(b *testing.B) {
...
-		sharedPVCsPerPod:       0,
-		uniquePVCsPerPod:       1,
+		nodes:                        500,
+		namespaces:                   200,
+		podsPerNode:                  200,
+		attachmentsPerNode:           20,
+		sharedConfigMapsPerNamespace: 0,
+		sharedConfigMapsPerPod:       0,
+		uniqueConfigMapsPerPod:       1,
+		sharedSecretsPerNamespace:    1,
```
Comment (liggitt) [inline plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: I think you can drop this... if `shared*PerNamespace` is < `shared*PerPod`, it uses the per-pod number, right?

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go [hunk 7]
```diff
@@ -840,71 +847,41 @@ func populate(graph *Graph, nodes []*corev1.Node, pods []*corev1.Pod, pvs []*cor
...
-func generate(opts sampleDataOpts) ([]*corev1.Node, []*corev1.Pod, []*corev1.PersistentVolume, []*storagev1.VolumeAttachment) {
+func generate(opts *sampleDataOpts) ([]*corev1.Node, []*corev1.Pod, []*corev1.PersistentVolume, []*storagev1.VolumeAttachment) {
 	nodes := make([]*corev1.Node, 0, opts.nodes)
 	pods := make([]*corev1.Pod, 0, opts.nodes*opts.podsPerNode)
 	pvs := make([]*corev1.PersistentVolume, 0, (opts.nodes*opts.podsPerNode*opts.uniquePVCsPerPod)+(opts.sharedPVCsPerPod*opts.namespaces))
 	attachments := make([]*storagev1.VolumeAttachment, 0, opts.nodes*opts.attachmentsPerNode)
 
+	rand.Seed(12345)
+	if opts.sharedSecretsPerNamespace < opts.sharedSecretsPerPod {
+		opts.sharedSecretsPerNamespace = opts.sharedSecretsPerPod
```
Comment (liggitt) [inline plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: not a fan of mutating the passed-in options here... either use `max(shared*PerPod, shared*PerNamespace)` at point of use or revert the options to a struct so changes here are local

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go (hunk 2 above)
Comment (liggitt) [reply plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: doc the behavior that `shared*PerPod` is used if greater than this number

Comment (wojtek-t) [reply plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: done

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go (hunk 6 above)
Comment (wojtek-t) [reply plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: done

File: plugin/pkg/auth/authorizer/node/node_authorizer_test.go (hunk 7 above)
Comment (wojtek-t) [reply plugin/pkg/auth/authorizer/node/node_authorizer_test.go]: done

**Review by wojtek-t — COMMENTED** [review]
Thanks for comments - PTAL

**Pull request merged**